File: DSA/PranyaGupta_500126032/Day6Question3.py
class Solution:
    def lengthSubarray(self, nums):
        if len(nums) == 0:
            return 0
        maxLen = 1
        currentLen = 1
        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1] + 1:
                currentLen += 1
                maxLen = max(maxLen, currentLen)
            else:
                currentLen = 1
        return maxLen

def main():
    user_input = input("Enter elements of the array (with space in between): ")
    nums = list(map(int, user_input.strip().split()))
    sol = Solution()
    result = sol.lengthSubarray(nums)
    print(result)

if __name__ == "__main__":
    main()

File: DSA/Nivedan_590014904/Day_6_Question_3.c
#include <stdio.h>
int maxsub(int arr[], int n) {
    int maxlen = 1, currlen = 1;
    for (int i = 1; i < n; i++) {
        if (arr[i] == arr[i - 1] + 1) {
            currlen++;
            if (currlen > maxlen) {
                maxlen = currlen;
            }
        } else {
            currlen = 1; 
        }
    }
    return maxlen;
}
int main() {
    int n;
    printf("Enter the number of elements: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter %d elements: ", n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }
    int result = maxsub(arr, n);
    printf("%d\n", result);
    return 0;
}
